server/models/meta.py

```diff
@@ -0,0 +1,263 @@
+        query = query or {}
+        query['tcga.type'] = self.TCGAType
+        return super(TCGAModel, self).find(query, **kwargs)
+
+    def findOne(self, query=None, **kwargs):
+        """Append TCGA model type to any query on this model."""
+        query = query or {}
+        query['tcga.type'] = self.TCGAType
+        return super(TCGAModel, self).findOne(query, **kwargs)
+
+    def setTCGA(self, doc, **tcga):
+        """Update the TCGA object and prune values of None."""
+        self.getTCGA(doc).update(tcga)
+        pruneNoneValues(self.getTCGA(doc))
+        return self
+
+    def getTCGA(self, doc):
+        """Return the TCGA object."""
+        return doc.setdefault('tcga', {})
+
+    def updateTCGAMeta(self, doc, meta):
+        """Update TCGA metadata."""
+        meta = updateDict(self.getTCGAMeta(doc), meta)
+        pruneNoneValues(meta)
+        return self
+
+    def getTCGAMeta(self, doc):
+        """Return TCGA metadata from a document."""
+        tcga = self.getTCGA(doc)
+        return tcga.setdefault('meta', {})
+
+    def removeTCGA(self, doc):
+        """Remove the tcga key and save the document.
+
+        This method will effectively reset the document so that
+        it no longer behaves as a TCGA document.  This is the
+        opposite of the import methods that promote Girder models
+        to TCGA types.
+        """
+        del doc['tcga']
+        self.save(doc, baseModel=True, validate=False)
+        return doc
+
+    def getTCGAType(self, doc):
+        """Get the type of model expressed by the document."""
+        return self.getTCGA(doc).get('type')
+
+    def getTCGACollection(self):
+        """Get the unique TCGA collection from the settings collection."""
+        tcga = self.model('setting').get(
+            TCGACollectionSettingKey
+        )
+        if tcga is None:
+            raise Exception(
+                'TCGA collection id not initialized in settings'
+            )
+        return self.model('collection').load(
+            tcga, force=True
+        )
+
+    def __upper(self, obj, key):
+        if key in obj:
+            obj[key] = obj[key].upper()
+
+    def __lower(self, obj, key):
+        if key in obj:
+            obj[key] = obj[key].lower()
+
+    def _parse(self, name, regex):
+        m = regex.match(name)
+        if m is None:
+            raise ValidationException('Invalid name')
+        d = m.groupdict()
+        self.__lower(d, 'uuid')
+        self.__upper(d, 'barcode')
+        self.__upper(d, 'case')
+        return d
+
+    def parseImage(self, name):
+        """Parse a slide image file name."""
+        return self._parse(name, self.image_re)
+
+    def parsePathology(self, name):
+        """Parse a pathology report file name."""
+        return self._parse(name, self.pathology_re)
+
+    @classmethod
+    def setMimeType(cls, doc):
+        """Set the mime type of a file document."""
+        doc['mimeType'] = mimetypes.guess_type(
```

Comment:
That's the way I [originally](https://github.com/DigitalSlideArchive/digital_slide_archive/pull/24/commits/3b5364963d0f36cac1185487f9d4edffe2d4de05) wrote it, but @scimk requested that we always override it because they had pdf's with type `application/octet-stream` from the original import.